### DISCUSSION TITLE
Icon Refresh

### DISCUSSION THREAD

**Discussion by andyholmes** [andyholmes/gnome-shell-extension-gsconnect#1483] [Polls]
@daudix-UFO, a member of the community, has proposed a new and updated icon for GSConnect.

Since GSConnect is a community-driven project, you have a chance to voice your opinion on the change!

## Version 1

![Version 1](https://user-images.githubusercontent.com/77155297/193826597-6ba2c335-911c-4151-83a1-e3f7a12ce2b2.png)

## Version 2

![Version 2](https://user-images.githubusercontent.com/77155297/194828649-e6bbb032-8e60-44f9-8da3-eda4a97a2b77.png)

## Version 3

![Version 3](https://user-images.githubusercontent.com/77155297/195461578-9365a502-beb7-4ff5-8918-178e1c6fdd0f.png)

## Version 4

![Version 4](https://user-images.githubusercontent.com/77155297/195462839-a53fe012-1a4b-4e3f-8f4c-4bf928f7f4de.png)

**Comment by siddhpant** [comment]
v1 looks like a BlackBerry phone.

v2's groove/cut emphasizes it is a laptop more than v1 IMHO, and looks fine as it is (though I would like if it was less subtle that an image was pasted over another).

Great job nevertheless!

**Reply by daudix** [reply]
> less subtle that an image was pasted over another

Can you explain what you mean by that 😅

**Reply by siddhpant** [reply]
The keyboard image being placed over the monitor...

I guess because I zoomed in I seem to now see it in that way (even if that wasn't the case, in which case I am extremely sorry).

![Screenshot_20221013-001729_Firefox](https://user-images.githubusercontent.com/25429745/195423754-36aeb21c-780c-431a-bda7-120c754a74a2.png)

**Reply by daudix** [reply]
It's not, just looks like for some reason xD

Keyboard:

![keyboard](https://user-images.githubusercontent.com/77155297/195424307-1e0eee54-c338-4b77-87c4-a908ef303577.png)

Monitor:

![monitor](https://user-images.githubusercontent.com/77155297/195424330-7394be6f-5f72-4db7-bb24-80d853311337.png)

Different colors to make it more visible:

![keyboard_monitor](https://user-images.githubusercontent.com/77155297/195424883-f1ff70ba-26b3-4148-b808-42045ad8c54e.png)

**Reply by ferdnyc** [reply]
@daudix-UFO 

I dunno... I still see what @siddhpant is saying. Even with the different color scheme, it _looks_ to me as if the keyboard is overlapping the screen.

I think part of it may be the shadow. Because the shadow is so dark (and the monitor frame is gray), the shadow _looks like_ a physical part of the monitor frame, rather than an actual shadow.

That aside, if the screen is hinged above the keyboard, the **screen** _should_ slightly overlap. Since the screen is (expected to be) slightly nearer to our POV, you wouldn't normally be able to see the very back edge of the keyboard -- the screen would slightly cover it. Either that, or it would be slightly higher than the base rather than touching it, with a small gap between them that's occupied only by the hinge(s). (Depends on the type of hinge used.)

So, the fact that we **can** see the keyboard's back edge, with the screen butted right up against it, creates the illusion that it's closer to us than the screen.

(I realize I'm late to this party, sorry about that.)

**Reply by ferdnyc** [reply]
(I just realized, maybe that gray isn't supposed to be a shadow at all, but rather the three-dimensional depth of the screen. In which case, it _definitely_ looks like it's overlapping, because from this perspective the **bottom** of the keyboard appears to be touching the **top** of the screen.)

**Reply by daudix** [reply]
The case is your second comment, it looks like this for a reason, in short, GNOME Guidelines don't allow making the depth in other direction and thinner than 2px, which makes it a bit hard to make 3d'is icon and not odd looking mess 😅

My idea was to make the screen look wide open, not 90° but rather 130°, unsuccessfully 😒

Your point is completely true, will note this in the future 😉

**Comment by daudix** [comment]
A bit less details, more modern laptop and phone

@andyholmes can you add this as variant 3/4?

![GSConnect_Alt2](https://user-images.githubusercontent.com/77155297/195461578-9365a502-beb7-4ff5-8918-178e1c6fdd0f.png)

![GSConnect_Alt3](https://user-images.githubusercontent.com/77155297/195462839-a53fe012-1a4b-4e3f-8f4c-4bf928f7f4de.png)

**Reply by rohmishra** [reply]
I like this design but a couple things stand out to me:
1. the chin on the cell phone looks out of place and kinda weird.
2. The keyboard buttons on 128x128 can be a bit smaller and tightly placed, currently it just looks weird. maybe adding a touchpad would be nice?

**Reply by daudix** [reply]
I tried all this and it not looked good 😔

**Comment by Arcitec** [comment]
Nice job. I prefer the 3rd for its nicer phone design and more modern laptop, but with the laptop keyboard size of the 2nd. Because the loose/wide laptop keys look really unbalanced and wrong.

**Reply by daudix** [reply]
I liked the old keyboard too, but it looks too detailed compared to other icons :(

**Reply by sharpenedblade** [reply]
The old keyboard is to detailed, but the revised keyboard doesnt look right, it might just be that people are used to a keyboard with more keys. The padding for the se3cond keyboard also looks off. I think a closed laptop might have less detail while also following the HIG better.

**Comment by amadeusp** [comment]
I think version 1 has the best contrast/background separation on dark mode. I think this is due to the thick bezel around the screen. Might not be the most modern looking but for the visibility/contrast my favorite.

**Comment by daudix** [comment]
Compromise between not very old hardware, visibility and metaphor

![Preview](https://user-images.githubusercontent.com/77155297/195735569-fd167393-6ae7-49c5-8e15-d5d2024afe2b.png)

**Reply by amadeusp** [reply]
Very good. IMHO it could be even better if the bezel was in a slight brighter grey (like on the initial version 1). What do you think?

**Reply by daudix** [reply]
Oops, picked wrong color 😅, thanks for noticing 🙃

**Reply by daudix** [reply]
Hmm, checked and the color is correct 🤔

This "optical illusion" can happen because black pixels are a lot and nearby to gray pixels

**Reply by sharpenedblade** [reply]
This version is pretty good, but I think the screen could be a little bigger. The edges on the bottom of the laptop also look triangular unless you look at it carefully, so maybe some changes there.

**Comment by andyholmes** [comment]
The new icon has been released :tada:

Thanks to @daudix-UFO  and all the votes and suggestions :slightly_smiling_face:

**Reply by daudix** [reply]
Thank you ❤ new quick settings UI looks fantastic!
